src/wq/ql/notify.py

```diff
@@ -129,30 +129,37 @@ def insert_task_finder(task_status: Optional[str] = None) \
 
 
 def list_insert_tasks(task_status: Optional[str] = None):
-    task_id_prefix = build_task_id_init_segment()
-    find_tasks = insert_task_finder(task_status)
-    response_payload = find_tasks(task_id_prefix)
-    log().info("Retrieved notification tasks to the work queue successfully")
-
-    return build_json_array_response_stream(response_payload)
-# TODO error handling
-
+    try:
+        task_id_prefix = build_task_id_init_segment()
+        find_tasks = insert_task_finder(task_status)
+        response_payload = find_tasks(task_id_prefix)
+        log().info("Retrieved notification tasks to the work queue successfully")
+        return build_json_array_response_stream(response_payload)
+    except Exception as e:
+        log().exception("list_insert_tasks failed: ", e)
+        raise e
 
-def list_insert_tasks_runtime_info():
-    task_id_prefix = build_task_id_init_segment()
-    response_payload = QMan.load_tasks_runtime_info(task_id_prefix)
-    log().info("Retrieved summary of notification tasks to the work queue successfully")
 
-    return build_json_array_response_stream(response_payload)
-# TODO error handling
+def list_insert_tasks_runtime_info(task_status: Optional[str] = None):
+    try:
+        task_id_prefix = build_task_id_init_segment()
+        response_payload = QMan.load_tasks_runtime_info(task_id_prefix)
+        log().info("Retrieved summary of notification tasks to the work queue successfully")
+        return build_json_array_response_stream(response_payload)
+    except Exception as e:
+        log().exception("list_insert_tasks_runtime_info failed: ", e)
+        raise e
 
 
-def delete_insert_tasks():
-    qman = QMan(InsertAction.insert_queue())
-    task_id_prefix = build_task_id_init_segment()
-    qman.delete_tasks(task_id_prefix)
-    log().info("Deleted notifcation tasks from the work queue successfully")
-# TODO error handling
+def delete_insert_tasks(task_status: Optional[str] = None):
+    try:
+        qman = QMan(InsertAction.insert_queue())
+        task_id_prefix = build_task_id_init_segment()
+        qman.delete_tasks(task_id_prefix)
+        log().info("Deleted notifcation tasks from the work queue successfully")
+    except Exception as e:
+        log().exception("delete_insert_tasks failed: ", e)
```

Comment:
this will cause a type error because `exception` doesn't expects an argument of type `Exception`. if you could please change the above line to:

```python
log().exception("delete_insert_tasks failed: ")
```